### PR TITLE
Changed http status code 404 to 204

### PR DESCRIPTION
**Pull request by L-Henke**
Client side implementation of Qabel/qabel-doc#82
Depends on Qabel/qabel-drop#15

### CODE REVIEW

**Comment by thechauffeur** [comment]
If Travis passes :+1: for merge.

File: src/test/java/de/qabel/core/http/DropHTTPTest.java [hunk 1]
```diff
@@ -229,7 +229,7 @@ public void shouldBeEmptyWithSinceDate() {
 		int responseCode = dHTTP.head(this.notExistingUrl,
```
Comment (aBothe) [inline src/test/java/de/qabel/core/http/DropHTTPTest.java]: notExistingUrl? This..should still be 404, right? Also, see the test comment which says `// HEAD 404 + SINCE`

Comment (cburkert) [inline src/test/java/de/qabel/core/http/DropHTTPTest.java]: No, there is no reason for 404 anymore. If the request is syntactically correct but the drop is emtpy then 204 is the right response, if the request is incorrect then 400 is.
So why did you change the assert but not the comment?

Comment (aBothe) [inline src/test/java/de/qabel/core/http/DropHTTPTest.java]: 400 stands for bad request, okay. But what if the URL is indeed valid but just not existing? That's 404 iirc.

Comment (cburkert) [inline src/test/java/de/qabel/core/http/DropHTTPTest.java]: You keep saying "does not exist", but in Qabel/qabel-core#82 we pointed out that a drop has no such property as existence.

Comment (aBothe) [inline src/test/java/de/qabel/core/http/DropHTTPTest.java]: Okay, not to spoil meta info. Okay, nice then :)
